pkg/autoscaler/statforwarder/forwarder.go

```diff
@@ -253,15 +274,80 @@ func (f *Forwarder) createOrUpdateEndpoints(ns, n string) error {
 	return nil
 }
 
-func (f *Forwarder) getHolder(name string) (string, bool) {
-	f.leaseLock.RLock()
-	defer f.leaseLock.RUnlock()
-	result, ok := f.leaseHolders[name]
+func (f *Forwarder) getProcessor(bkt string) (*bucketProcessor, bool) {
+	f.processorsLock.RLock()
+	defer f.processorsLock.RUnlock()
+	result, ok := f.processors[bkt]
 	return result, ok
 }
 
-func (f *Forwarder) setHolder(name, holder string) {
-	f.leaseLock.Lock()
-	defer f.leaseLock.Unlock()
-	f.leaseHolders[name] = holder
+func (f *Forwarder) setProcessor(bkt string, p *bucketProcessor) {
+	f.processorsLock.Lock()
+	defer f.processorsLock.Unlock()
+	f.processors[bkt] = p
+}
+
+func (f *Forwarder) createProcessor(bkt, holder string) *bucketProcessor {
+	if holder == f.selfIP {
+		return &bucketProcessor{
+			holder: holder,
+			proc: func(sm asmetrics.StatMessage) {
+				l := f.logger.With(zap.String("revision", sm.Key.String()))
+				l.Debugf("Accept stat as owner of bucket %s", bkt)
```

Comment:
```suggestion
				l.Debug("Accept stat as owner of bucket ", bkt)
```
Since it's a single usage logger, perhaps not even create the variable but just join the two lines